flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -1040,6 +1066,32 @@ void createTemporarySystemFunction(
      */
     boolean dropTemporaryView(String path);
 
+    /**
+     * Drops a view registered in the given path.
+     *
+     * <p>Temporary objects can shadow permanent ones. If a temporary object exists in a given path,
+     * make sure to drop the temporary object first using {@link #dropTemporaryView}. This method
```

Comment:
nit: I would move sentence "This method can only drop permanent objects" to the beginning of the paragraph. So it is obvious what this method is before we talk about temporary objects.